[IR2VEC] Rewrite triples.mir and entities.mir. NFC

davemgreen opened the pull request:
These test change and need updating every time a new instruction is added, including generic G_ global isel instructions. The opcodes of all instructions change force a manual update of the file, which makes it very high cost for what it is testing.

I have rewritten it to use regexes for the triplets.mir test and just test a couple of opcodes in the entities.mir file.

svkeerthy reviewed (approved):
Thanks for the patch. We kept it this way to track any changes in the IR. We now plan to setup buildbot to do this. Current changes look good to me.

davemgreen committed:
[IR2VEC] Rewrite triples.mir and entities.mir. NFC

These test change and need updating every time a new instruction is added,
including generic G_ global isel instructions. The opcodes of all instructions
change force a manual update of the file, which makes it very high cost for
what it is testing.

I have rewritten it to use regexes for the triplets.mir test and just test a
couple of opcodes in the entities.mir file.

Also change them from REQUIRES: x86_64-linux to REQUIRES:
x86-registered-target, as they should run fine on other targets providing that
the triple is used correctly. It passes successfully on an AArch64 box I tried
it on.
